### ISSUE TITLE
Version 6.2.0 type hinting breaks mypy with --strict

### ISSUE THREAD

**Issue by spackard** [jd/tenacity#230]
With the following simple program, running `mypy --strict` claims the methods are not part of the tenacity module. Amusingly, mypy suggests other names in the tenacity module, and if you add more code using those names, mypy complains about them also and suggests the original names it complained about. Adding `# type: ignore` comments does silence the errors, but that's not ideal.

```python
import tenacity


@tenacity.retry(
    retry=tenacity.retry_if_exception_type(RuntimeError),
    wait=tenacity.wait_exponential(exp_base=1.2, multiplier=1 / 1.2),
    stop=tenacity.stop_after_attempt(10),
    reraise=True,
)
def keep_trying() -> None:
    raise RuntimeError("Oops")


keep_trying()
```

```console
$ mypy --strict main.py
main.py:5: error: Module has no attribute "retry_if_exception_type"; maybe "retry_if_exception", "retry_unless_exception_type", or "retry_if_exception_message"?
main.py:6: error: Module has no attribute "wait_exponential"; maybe "wait_random_exponential"?
main.py:7: error: Module has no attribute "stop_after_attempt"
Found 3 errors in 1 file (checked 1 source file)
```

**Comment by GPHemsley** [comment]
Note that I filed python/mypy#8754 for this same issue.

The workaround reported there to silence these errors is to allow implicit re-exports with `--implicit-reexport`.

**Comment by spackard** [comment]
Ah, I think in `__init__.py`, if you change this:

```python
from .retry import retry_all
```

to this:

```python
from .retry import retry_all as retry_all
```

ie change the implicit import into an explicit import, the tool stops complaining. We do that in our internal code base after mypy included no implicit imports as part of strict.

**Comment by JenSte** [comment]
In a local copy of tenacity I changed the imports to the `from <module> import <symbol> as <symbol>` form so mypy can find the functions, however it now complains with `error: Call to untyped function "<functionname>" in typed context`.

Currently tenacity includes a `py.typed` file, but only contains type hints for a small number of functions. IMHO this file should be removed again until all the code has type annotations.

**Comment by stevenpackardblp** [comment]
Would it be correct to say #315 fixed this issue?

**Comment by stevenpackardblp** [comment]
Hmm, looks like it may not have completely covered it:

```console
error: "stop_base" has no attribute "max_attempt_number"
```

**Comment by bhrutledge** [comment]
I'm still getting this error w/ `tenacity==8.0.1`:

> Module has no attribute "retry_if_exception_type"; maybe "retry_if_not_exception_type", "retry_if_exception", or "retry_unless_exception_type"?

I think #347 might fix this. Until then, I can avoid the error by enabling implicit re-exports for tenacity, e.g. in `.mypy.ini`:

```
[mypy-tenacity.*]
implicit_reexport = True
```

**Comment by sileht** [comment]
This fixed by #347
